UNSTATED_3_RFC.md

```diff
@@ -0,0 +1,138 @@
+# Unstated 3.0 RFC
+
+I wanted to create an RFC for Unstated 3.0 which I want to take advantage of
+the new [React Hooks](https://reactjs.org/docs/hooks-intro.html) APIs.
+
+## Example
+
+My idea for it right now looks like this:
+
+```js
+import React, { useState } from 'react';
+import { Provider, useProvided, useInject } from 'unstated';
+
+function useCounter(initialCount = 0) {
+  let [count, setCount] = useState(initialCount);
+  let reset = () => setCount(0);
+  let increment = () => setCount(count => count - amount);
+  let decrement = () => setCount(count => count + amount);
+  return { count, reset, increment, decrement };
+}
+
+function Counter() {
+  // use existing instance from context, or create new one
+  let counter = useProvided(useCounter); 
+  return <>
+    Count: {counter.count}
+    <button onClick={counter.reset}>Reset</button>
+    <button onClick={counter.increment}>+</button>
+    <button onClick={counter.decrement}>-</button>
+  </>;
+}
+
+function App() {
+  // create new instance of container
+  let counter = useInject(useCounter, 42); 
```

Comment:
I really liked the auto inject in current. Just need 1 `Provider` at the root and forget. Would this mean you have to inject all context before using them?

Reply:
No... it's exactly the same